File: p1068.cpp
#include<bits/stdc++.h>
using namespace std;
#define MAX 54
int N, parent, root, del;
vector<int> MAP[MAX];
int go(int node){
    int ret = 0;
    int childCnt = 0;
    for(int i = 0; i < MAP[node].size(); i++){
        if(MAP[node][i] == del) continue;
        ret += go(MAP[node][i]);
        childCnt++;
    }
    if(childCnt == 0) return 1;
    else return ret;
}
int main(){
    cin >> N;
    for(int i = 0; i < N; i++){
        cin >> parent;
        if(parent == -1) root = i;
        else MAP[parent].push_back(i);
    }
    cin >> del;
    if(del == root) cout << 0 << "\n";
    else cout << go(root) << "\n";
}

File: p1068.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.ArrayList;
import java.util.StringTokenizer;

public class p1068 {
    public static int N;
    public static int root;
    public static int del;
    public static ArrayList<Integer> tree[];
    public static void main(String[] args) throws IOException{
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        N = Integer.parseInt(br.readLine());
        tree = new ArrayList[N];
        for(int i = 0; i < N; i++) tree[i] = new ArrayList<Integer>();

        StringTokenizer st = new StringTokenizer(br.readLine());
        for(int i = 0; i < N; i++){
            int parent = Integer.parseInt(st.nextToken());
            if(parent == -1) root = i;
            else tree[parent].add(i);
        }
        del = Integer.parseInt(br.readLine());
        
        System.out.println((del == root) ? 0 : dfs(root));
    }

    public static int dfs(int idx){
        if(tree[idx].isEmpty() || tree[idx].size() == 1 && tree[idx].get(0) == del) return 1;
        int ret = 0;
        for(int i = 0; i < tree[idx].size(); i++){
            if(tree[idx].get(i) != del) ret += dfs(tree[idx].get(i));
        }
        return ret;
    }
}
